implementation/src/main/java/io/smallrye/config/KeyMap.java

```diff
@@ -250,4 +251,22 @@ public StringBuilder toString(StringBuilder b) {
     public String toString() {
         return toString(new StringBuilder()).toString();
     }
+
+    public Map<String, V> toMap() {
```

Comment:
This will tend to result in a wrong answer because it leaves out every "any" value.

Reply:
I'll check.

Reply:
IMO this is still a very deceptive method.  There is no equivalency between KeyMap and Map, the former being hierarchical and pattern based, and the latter being flat.  This will only lead to temptation to implement things incorrectly (like my comment about the incorrectness of the KeyMap-backed config source).